### PR TITLE
added support tethys json

### PR DESCRIPTION
**Pull request by chernykhSG**
Before submitting pull request:
- [x] Check if the project compiles by running `sbt compile`
- [x] Verify docs compilation by running `sbt compileDocs`
- [x] Check if tests pass by running `sbt test`
- [x] Format code by running `sbt scalafmt`

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -843,6 +843,23 @@ lazy val zio1Json = (projectMatrix in file("json/zio1-json"))
   .jsPlatform(scalaVersions = scala2 ++ scala3, settings = commonJsSettings)
   .dependsOn(core, jsonCommon)
 
+lazy val tethysJson = (projectMatrix in file("json/tethys-json"))
```
Comment (adamw) [inline build.sbt:846]: this should also be added to `allAggregates`

File: build.sbt [hunk 2]
```diff
@@ -843,6 +843,23 @@ lazy val zio1Json = (projectMatrix in file("json/zio1-json"))
...
+  .settings(
+    name := "tethys-json",
+    libraryDependencies ++= Seq(
+      "com.tethys-json" %% "tethys-core" % "0.26.0",
```
Comment (adamw) [inline build.sbt:850]: maybe define a constant in the top of the build file with the version?

File: build.sbt [hunk 3]
```diff
@@ -843,6 +843,23 @@ lazy val zio1Json = (projectMatrix in file("json/zio1-json"))
...
+      "com.tethys-json" %% "tethys-jackson" % "0.26.0",
+      "com.tethys-json" %% "tethys-derivation" % "0.26.0"
+    ),
+    scalaTest
+  )
+  .jvmPlatform(
+    scalaVersions = Seq(scala2_12, scala2_13),
```
Comment (adamw) [inline build.sbt:857]: no scala 3 support?

File: build.sbt (hunk 2 above)
Comment (chernykhSG) [reply build.sbt:850]: There is support, but the new version only supports 3 scala; for 3 scala you need to make a separate implementation.

File: build.sbt (hunk 3 above)
Comment (chernykhSG) [reply build.sbt:857]: There is support, but the new version only supports 3 scala; for 3 scala you need to make a separate implementation.

File: build.sbt (hunk 1 above)
Comment (chernykhSG) [reply build.sbt:849]: corrected

File: build.sbt (hunk 3 above)
Comment (chernykhSG) [reply build.sbt]: added support scala3

**Comment by adamw** [comment]
Thanks :)

**Pull request merged**